src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -5095,6 +5095,22 @@ void CEEInfo::getCallInfo(
         EX_THROW(EEMessageException, (kMissingMethodException, IDS_EE_MISSING_METHOD, W("?")));
     }
 
+    // If this call is for a LDFTN and the target method has the NativeCallableAttribute,
+    // then validate it adheres to the limitations.
+    if ((flags & CORINFO_CALLINFO_LDFTN) && pMD->HasNativeCallableAttribute())
+    {
+        if (!pMD->IsStatic())
```

Comment:
Can we delete this check from ``COMDelegate::ConvertToCallback` ?

Reply:
The `COMDelegate::ConvertToCallback` is called in a scenario below. However, I think that should be updated as well right?

https://github.com/dotnet/runtime/blob/ecd8378926fa27c5ee838ff0c5e67798f88e127e/src/coreclr/src/vm/jitinterface.cpp#L13353-L13361

Reply:
Right